doc/en/user/source/security/passwd.rst

```diff
@@ -97,8 +97,14 @@ By default, the master password is generated and stored in a file named ``securi
 
 .. warning:: The file ``security/masterpw.info`` is a security risk. The administrator should read this file and verify the master password by logging on GeoServer as the ``root`` user. On success, this file should be removed.
 
+.. warning:: The first thing an Administrator of the System should do is dump the Master Password generated by GeoServer, store it somewhere not accessible by anyone, and delete ``security/masterpw.info`` or whatever file you dumped the password to.
+
 Refer to :ref:`security_webadmin_masterpasswordprovider` for information on how to change the master password.
 
+.. note:: By default login to the Admin GUI and REST APIs using the Master Password is disabled. In order to enable it you will need to manually change the Master Password Provider config.xml, usually located in ``security/masterpw/default/config.xml``, by adding the following statement::
```

Comment:
Nope, the intent is exactly to disable the admin GUI by default, in order to avoid security issues. Generated master passwords are not randomized, therefore the idea here is to make an administrator aware of what he is doing.

Consider that if the administrator does not dump the master password in clear explicitly, he won't be able to access the GUI anyway. So basically this change does not change the current behavior.

Reply:
Then please fix the title and jira ticket, they are both mis-representing what this does (it's actually a backwards incompatible change that may bite unaware people on upgrade, so needs to be advertised prominently on release notes).

Reply:
Also, I see no GUI way to enable back the master password, which is making re-enabling it overly complicated when running from a remote host (I certainly would not start fiddling on the security config from remote without a master password enabled). Can you please add a checkbox?

Reply:
@aaime done!